### PR TITLE
Dtspo4393 adding route table

### PR DESCRIPTION
**Pull request by darren221980**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-4393

### Change description ###

Adding route table creation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: components/network/init.tf [hunk 1]
```diff
@@ -75,3 +75,24 @@ provider "azurerm" {
   features {}
   alias = "vpn"
 }
+
+provider "azurerm" {
+
+  alias                      = "core-infra-routetable"
```
Comment (thomast1906) [inline components/network/init.tf:81]: Formatting/spacing - can you resolve

File: components/network/inputs-default.tf [hunk 2]
```diff
@@ -93,4 +93,20 @@ variable "product" {
 
 variable "additional_routes_appgw" {
   default = []
+}
+
+variable "route_name" {
+  default = "default"
+}
+
+variable "route_address_prefix" {
```
Comment (thomast1906) [inline components/network/inputs-default.tf:102]: Is this the required route setup?

Example of route table you are looking to automate
https://portal.azure.com/#@HMCTS.NET/resource/subscriptions/1c4f0704-a29e-403d-b719-b90c34ef14c9/resourceGroups/core-infra-aat/providers/Microsoft.Network/routeTables/aks-aat-core-infra-route-table/overview

File: components/network/routetable.tf [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  name                   = var.route_name
+  route_table_name       = azurerm_route_table.route_table.name
+  resource_group_name    = "core-infra-${var.environment}"
+  address_prefix         = var.route_address_prefix
+  next_hop_type          = var.route_next_hop_type
+  next_hop_in_ip_address = var.route_next_hop_in_ip_address
+
+  provider = azurerm.core-infra-routetable
+}
+
```
Comment (thomast1906) [inline components/network/routetable.tf:24]: Do we need both resources? Can we look at 1 x for_each that we can amend to

File: components/network/init.tf (hunk 1 above)
Comment (darren221980) [reply components/network/init.tf]: fixed

File: components/network/routetable.tf (hunk 3 above)
Comment (darren221980) [reply components/network/routetable.tf]: now only 1 resource using for_each

File: environments/network/ithc.tfvars [hunk 4]
```diff
@@ -18,7 +18,19 @@ private_dns_zones = [
 
 hub = "nonprod"
 
-additional_routes = [
```
Comment (thomast1906) [inline environments/network/ithc.tfvars:21]: As discussed, should not be removing any routetables/routes

File: components/network/inputs-default.tf (hunk 2 above)
Comment (thomast1906) [reply components/network/inputs-default.tf]: Resolved

**Review by chrisorisawayi — APPROVED** [review]

**Pull request merged**